sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereQuery.cs

```diff
@@ -94,8 +116,70 @@ public WhereQuery WhereStartsWith(string stringToCheck, string beginningString)
         /// <returns> ADT query that already contains SELECT and FROM. </returns>
         public WhereQuery WhereIsOfModel(string model, bool exact = false)
         {
-            Console.WriteLine(model);
-            Console.WriteLine(exact);
+            if (exact)
+            {
+                _clauses.Add(new WhereClause($"{QueryConstants.IsOfModel}('{model}', exact)"));
+            }
+            else
+            {
+                _clauses.Add(new WhereClause($"{QueryConstants.IsOfModel}('{model}')"));
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the <see href="https://docs.microsoft.com/en-us/azure/digital-twins/reference-query-functions#is_bool">IS_BOOL</see> function to the condition statement of the query.
+        /// </summary>
+        /// <param name="expression"> The expression that the query is looking for as boolean. </param>
+        /// <returns> ADT query that already contains SELECT and FROM. </returns>
+        public WhereQuery WhereIsBool(string expression)
```

Comment:
for these type check kinds of conditions, you can group them together and use an enum value for the type. I am not sure if `expression` is the best parameter name either ... what goes for the expression? is it usually a property?

maybe: 
`WhereIsOfType(string property, AdtDatatype type)`

Reply:
and maybe you define AdtDataType enum to have `Bool`, `Number`, `String` , `Primitive` , `Object` as different options.